j5/backends/backend.py

```diff
@@ -173,3 +173,28 @@ def get_backend(self, board: Type['Board']) -> Type[Backend]:
             raise NotImplementedError(f"The {str(self)} does not support {str(board)}")
 
         return self.board_backend_mapping[board]
+
+    def merge(self, other: 'Environment') -> None:
+        """
+        Merge in the board-backend mappings from another environment.
+
+        This allows vendors to predefine Environments and API authors
+        can then merge several vendor environments to get the one that
+        they need for their API.
+
+        This method will fail if any board is defined in both environments,
+         as it is unclear which one has the correct mapping.
+        """
+        boards_present = self.supported_boards | other.supported_boards
+        total_boards = len(self.supported_boards) + len(other.supported_boards)
+
+        if len(boards_present) != total_boards:
```

Comment:
Would it be simpler to test that the intersection (which you compute below anyway) is non-empty?

Reply:
0a525ef